src/api/ledger/parse/pathfind.js

```diff
@@ -8,17 +8,25 @@ function parsePaths(paths) {
     _.omit(step, ['type', 'type_hex'])));
 }
 
+function ensureIOUCounterparty(amount, sourceAmount) {
+  const defaults = sourceAmount.counterparty ?
+    {counterparty: sourceAmount.counterparty} : {};
+  return amount.currency === 'XRP' || amount.counterparty ? amount :
+    _.assign(defaults, amount);
+}
+
 function parsePathfind(sourceAddress: string,
     destinationAmount: Object, pathfindResult: Object): Object {
   return pathfindResult.alternatives.map(function(alternative) {
+    const sourceAmount = parseAmount(alternative.source_amount);
     return {
       source: {
         address: sourceAddress,
-        amount: parseAmount(alternative.source_amount)
+        amount: sourceAmount
       },
       destination: {
         address: pathfindResult.destination_account,
-        amount: destinationAmount
+        amount: ensureIOUCounterparty(destinationAmount, sourceAmount)
```

Comment:
"Never be exposed" refers to this part "When the issuer field of the destination Amount field matches the Destination account address, it is treated as a special case meaning “any issuer that the destination accepts.""

If you interpret it literally, it's actually _wrong_ because it is not an IOU issued by the destination, that's just a special code. A user of the new API should never see that in the requests of responses.